lib/datadog/core/configuration/components.rb

```diff
@@ -345,6 +355,9 @@ def build_profiler_scheduler(_settings, exporter, transport)
           :runtime_metrics,
           :tracer
 
+        # Telemetry instances persist across multiple component restarts
+        attr_accessor :telemetry
```

Comment:
Because you have an accessor for telemetry, you should be able to do `Datadog.send(:components).telemetry.enabled` in the integration tests above.